tanjun/abc.py

```diff
@@ -318,26 +318,87 @@ def set_component(self: _T, _: typing.Optional[Component], /) -> _T:
 
     @abc.abstractmethod
     async def fetch_channel(self) -> hikari.PartialChannel:
+        """Fetch the channel the context was invoked in.
+
+        !!! note
+            This performs an API call. Consider using `Context.get_channel`
+            if you have `hikari.config.CacheComponents.GUILD_CHANNELS` cache component enabled.
+
+        Returns
+        -------
+        hikari.PartialChannel
+            The guild channel the context was invoked in.
```

Comment:
Just a more general note for all of these, but a Raises segment would be helpful for these and hikari's docs can be used as a ref for that